src/DiagonalHessianApproximation.jl

```diff
@@ -70,15 +77,33 @@ function push!(
   trA2 = dot(s2, s2)
   sT_y = dot(s, y)
   sT_B_s = dot(s2, B.d)
+  sT_y0 = dot(s0, y0)
+  sT_B_s0 = dot(s0.^2, B.d)
   q = sT_y - sT_B_s
   if B.psb
     q /= trA2
     B.d .+= q .* s .^ 2
-  else
+  elseif B.andrei
     sT_s = dot(s, s)
     q += sT_s
     q /= trA2
     B.d .+= q .* s .^ 2 .- 1
+  elseif B.wolk
+    @assert all(x -> x > 0, B.d)
+    σ1 = dot(s0.^2, sqrt.(B.d))
```

Comment:
Is this line allocating? If it is, you might be able to use something like `s2 = (si^2 for si ∈ s)` a few lines above. You can check the allocations with `@allocated`.